tsfresh/feature_extraction/extraction.py

```diff
@@ -214,171 +190,112 @@ def _extract_features_per_kind(kind_to_df_map,
-                              total=total_number_of_expected_results,
-                              desc="Feature Extraction", disable=disable_progressbar)
-    pool.close()
-
-    # Concatenate all partial results
-    result = pd.concat(extracted_features, axis=1, join='outer').astype(np.float64)
+    extraction_function = partial(_do_extraction_on_chunk,
+                                  default_fc_parameters=default_fc_parameters,
+                                  kind_to_fc_parameters=kind_to_fc_parameters)
 
-    # Impute the result if requested
-    if impute_function is not None:
-        impute_function(result)
+    # Map over all those chunks and extract the features on them
+    result = tqdm(map_function(extraction_function, data_in_chunks),
+                  total=total_number_of_expected_results,
+                  desc="Feature Extraction", disable=disable_progressbar)
 
-    pool.join()
-    return result
+    # Flatten out the lists
+    result = itertools.chain.from_iterable(result)
 
+    # Return a dataframe in the typical form (id as index and feature names as columns)
+    result = pd.DataFrame(list(result), dtype=np.float)
 
-def _extract_features_parallel_per_sample(kind_to_df_map,
-                                          column_id, column_value,
-                                          default_fc_parameters,
-                                          kind_to_fc_parameters=None,
-                                          chunksize=defaults.CHUNKSIZE,
-                                          n_processes=defaults.N_PROCESSES, show_warnings=defaults.SHOW_WARNINGS,
-                                          disable_progressbar=defaults.DISABLE_PROGRESSBAR,
-                                          impute_function=defaults.IMPUTE_FUNCTION):
-    """
-    Parallelize the feature extraction per kind and per sample.
+    if len(result) != 0:
+        result = result.pivot("id", "variable", "value")
 
-    As the splitting of the dataframes per kind along column_id is quite costly, we settled for an async map in this
-    function. The result objects are temporarily stored in a fifo queue from which they can be retrieved in order
-    of submission.
+    return result
 
-    :param kind_to_df_map: The time series to compute the features for in our internal format
-    :type kind_to_df_map: dict of pandas.DataFrame
 
-    :param column_id: The name of the id column to group by.
-    :type column_id: str
+def _do_extraction_on_chunk(chunk, default_fc_parameters, kind_to_fc_parameters):
+    """
+    Main function of this module: use the feature calculators defined in the
+    default_fc_parameters or kind_to_fc_parameters parameters and extract all
+    features on the chunk.
 
-    :param column_value: The name for the column keeping the value itself.
-    :type column_value: str
+    The chunk consists of the chunk id, the chunk kind and the data (as a Series),
```

Comment:
Well, actually I do not know. ID can be anything (pandas allows ints as well as strings or other objects) and the kind can also be anything (even an array).